src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -46,19 +54,50 @@ class KubeTransferStrategy implements TransferStrategy {
     @Inject
     private K8sService k8sService
 
+    @Inject
+    private CleanupStrategy cleanup
+
+    @Inject
+    @Named(TaskExecutors.IO)
+    private ExecutorService executor
+
     @Override
     BlobCacheInfo transfer(BlobCacheInfo info, List<String> command) {
-        final podName = podName(info)
-        final pod = k8sService.transferContainer(podName, blobConfig.s5Image, command, blobConfig)
-        final terminated = k8sService.waitPod(pod, blobConfig.transferTimeout.toMillis())
-        final stdout = k8sService.logsPod(podName)
-        return terminated
+        final name = getName(info)
+        final job = k8sService.transferJob(name, blobConfig.s5Image, command, blobConfig)
+        final podList = k8sService.waitJob(job, blobConfig.transferTimeout.toMillis())
+        final size = podList.items.size()
+
+        if( size < 1 )
+            throw new TransferTimeoutException("Transfer job timed out")
+
+        // Find the latest created pod among the pods associated with the job
+        def latestPod = podList.getItems().get(0)
+        for (def pod : podList.items) {
+            if (pod.metadata.creationTimestamp.isAfter(latestPod.metadata.creationTimestamp)) {
+                latestPod = pod
+            }
+        }
+
```

Comment:
I do not understand this logic. I believe when using job to run the transfer also the wait should be done against the job status not the pod. Otherwise, in the case of a retry event, the wait condition will be inconsistent because a new pod will be created

Reply:
ohhh, got it now, ignore the above comment